### PR TITLE
[SYCL] Fixed return of uninitialized cl_program from PI

### PR DESCRIPTION
**Pull request by KarachunIvan**
Signed-off-by: Ivan Karachun <ivan.karachun@intel.com>

### CODE REVIEW

File: sycl/source/detail/pi_opencl.cpp [hunk 1]
```diff
@@ -153,10 +155,11 @@ pi_program OCL(piProgramCreate)(pi_context context, const void *il,
           curPlatform, "clCreateProgramWithILKHR"));
 
   assert(funcPtr != nullptr);
-  resProgram = funcPtr(pi_cast<cl_context>(context), il, length,
-                         pi_cast<cl_int *>(err));
+    if (res_program != nullptr)
+      *res_program = pi_cast<pi_program>(funcPtr(pi_cast<cl_context>(context),
+                         il, length, pi_cast<cl_int *>(&err)));
```
Comment (bader) [inline sycl/source/detail/pi_opencl.cpp]: Please, apply clang-format to you patch.

Comment (KarachunIvan) [reply sycl/source/detail/pi_opencl.cpp]: Done

File: sycl/source/detail/program_manager/program_manager.cpp [hunk 2]
```diff
@@ -78,10 +78,10 @@ static RT::PiProgram createSpirvProgram(const RT::PiContext Context,
...
                                         size_t DataLen) {
   RT::PiResult Err = PI_SUCCESS;
-  RT::PiProgram Program;
-  PI_CALL((Program = pi::pi_cast<pi_program>(
+  RT::PiProgram Program = nullptr;
+  PI_CALL((Err = pi::pi_cast<pi_result>(
                pi::piProgramCreate(pi::pi_cast<pi_context>(Context), Data, DataLen,
-                                   pi::pi_cast<pi_result *>(&Err))),
+                                   pi::pi_cast<pi_program *>(&Program))),
            Err));
```
Comment (smaslov-intel) [inline sycl/source/detail/program_manager/program_manager.cpp]: this should be just:

PI_CALL(pi::piProgramCreate(Context, Data, DataLen, &Program));

Comment (KarachunIvan) [reply sycl/source/detail/program_manager/program_manager.cpp]: done

**Comment by bader** [comment]
@KarachunIvan, all commits must be signed.

**Pull request merged**